File: miracle-api/src/main/java/com/depromeet/service/record/RecordRetrieveService.java
package com.depromeet.service.record;

import com.depromeet.domain.record.Record;
import com.depromeet.domain.record.RecordRepository;
import com.depromeet.service.record.dto.request.DayRecordsRequest;
import com.depromeet.service.record.dto.request.MonthRecordsRequest;
import com.depromeet.service.record.dto.response.RecordResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

@RequiredArgsConstructor
@Service
public class RecordRetrieveService {

    private final RecordRepository recordRepository;

    @Transactional(readOnly = true)
    public List<RecordResponse> retrieveMonthRecords(MonthRecordsRequest request, Long memberId) {
        List<Record> records = recordRepository.findRecordBetween(memberId, request.convertFirstDayOfTheMonth(), request.convertEndDayOfTheMonth());
        return records.stream()
            .map(RecordResponse::of)
            .collect(Collectors.toList());
    }

    @Transactional(readOnly = true)
    public List<RecordResponse> retrieveDayRecords(DayRecordsRequest request, Long memberId) {
        List<Record> records = recordRepository.findRecordBetween(memberId, request.convertMinLocalDateTime(), request.convertMaxLocalDateTime());
        return records.stream()
            .map(RecordResponse::of)
            .collect(Collectors.toList());
    }

}

File: miracle-api/src/main/java/com/depromeet/service/record/RecordRetrieveService.kt
package com.depromeet.service.record

import com.depromeet.domain.record.Record
import com.depromeet.domain.record.RecordRepository
import com.depromeet.service.record.dto.request.DayRecordsRequest
import com.depromeet.service.record.dto.request.MonthRecordsRequest
import com.depromeet.service.record.dto.response.RecordForCalendarResponse
import com.depromeet.service.record.dto.response.RecordResponse
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.stream.Collectors

@Service
class RecordRetrieveService(private val recordRepository: RecordRepository) {

    @Transactional(readOnly = true)
    fun retrieveMonthRecords(request: MonthRecordsRequest, memberId: Long): RecordForCalendarResponse {
        val records = recordRepository.findRecordBetween(
            memberId,
            request.convertFirstDayOfTheMonth(),
            request.convertEndDayOfTheMonth()
        )
        return RecordForCalendarResponse.of(records)
    }

    @Transactional(readOnly = true)
    fun retrieveDayRecords(request: DayRecordsRequest, memberId: Long): List<RecordResponse> {
        val records = recordRepository.findRecordBetween(
            memberId,
            request.convertMinLocalDateTime(),
            request.convertMaxLocalDateTime()
        )
        return records.stream()
            .map { record: Record ->
                RecordResponse.of(
                    record
                )
            }
            .collect(Collectors.toList())
    }
}
